### PR TITLE
Splash Screen feature

### PR DESCRIPTION
**Pull request by SebasAMont**
Implementation of SplashScreen and the necessary tests. 

 

https://user-images.githubusercontent.com/109122811/195408969-78030ff4-0d0b-4598-8162-38fc2761a058.mp4

 
image

### CODE REVIEW

File: test/widget_tests/splash_screen_test.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  Widget buildWidget() {
+    return MaterialApp(
+      home: const SplashScreen(
+        key: Key('screenKey'),
+      ),
+      routes: {
+        'SplashScreen': (context) => const SplashScreen(),
+        'MainScreen': (context) => const MainScreen(),
```
Comment (GARAF004) [inline test/widget_tests/splash_screen_test.dart:28]: Add routes to route constants file

File: test/widget_tests/splash_screen_test.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  late MockNavigatorObserver mockNavObserver;
+
+  setUp(() {
+    mockNavObserver = MockNavigatorObserver();
+  });
+
+  Widget buildWidget() {
+    return MaterialApp(
+      home: const SplashScreen(
+        key: Key('screenKey'),
```
Comment (GARAF004) [inline test/widget_tests/splash_screen_test.dart:24]: screenKey? This key is more related to splash

File: test/widget_tests/splash_screen_test.dart (hunk 1 above)
Comment (FranciscoGaraventa) [inline test/widget_tests/splash_screen_test.dart]: Add this routes to a route constant file

File: test/widget_tests/splash_screen_test.dart (hunk 2 above)
Comment (FranciscoGaraventa) [inline test/widget_tests/splash_screen_test.dart]: screenKey? This key i think is more related to Splash

File: test/widget_tests/splash_screen_test.dart (hunk 1 above)
Comment (SebasAMont) [reply test/widget_tests/splash_screen_test.dart]: Done.

File: test/widget_tests/splash_screen_test.dart (hunk 2 above)
Comment (SebasAMont) [reply test/widget_tests/splash_screen_test.dart]: screenKey is not used so it will be removed. Done

**Review by MarcosCGlobant — APPROVED** [review]

**Review by FranciscoGaraventa — APPROVED** [review]

**Pull request merged**